src/app/view/perspectives/table/header/column-group/single-column/input/table-column-input.component.ts

```diff
@@ -0,0 +1,122 @@
+ * Copyright (C) since 2017 Lumeer.io, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  SimpleChanges,
+  ViewChild,
+} from '@angular/core';
+import {KeyCode} from '../../../../../../../shared/key-code';
+import {HtmlModifier} from '../../../../../../../shared/utils/html-modifier';
+
+const DISABLED_CHARACTERS = ['.'];
+
+@Component({
+  selector: 'table-column-input',
+  templateUrl: './table-column-input.component.html',
+  styleUrls: ['./table-column-input.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TableColumnInputComponent implements OnChanges {
+  @Input()
+  public default: boolean;
+
+  @Input()
+  public edited: boolean;
+
+  @Input()
+  public initialized: boolean;
+
+  @Input()
+  public value: any;
+
+  @Output()
+  public valueChange = new EventEmitter<string>();
+
+  @Output()
+  public save = new EventEmitter<string>();
+
+  @Output()
+  public cancel = new EventEmitter();
+
+  @ViewChild('textInput', {static: false})
+  public textInput: ElementRef<HTMLInputElement>;
+
+  public ngOnChanges(changes: SimpleChanges) {
+    if (changes.edited && this.edited) {
+      setTimeout(() => this.focusInput());
+    }
+  }
+
+  private focusInput() {
+    const element = this.textInput && (this.textInput.nativeElement as HTMLElement);
+    if (element) {
+      HtmlModifier.setCursorAtTextContentEnd(element);
+      element.focus();
+    }
+  }
+
+  public onBlur() {
+    if (this.initialized) {
+      this.cancel.emit();
+    } else {
+      this.saveValue();
+    }
+  }
+
+  public onInput(event: Event) {
+    const value = event.target['value'] || '';
+    this.valueChange.emit(value.trim());
+  }
+
+  public onKeyDown(event: KeyboardEvent) {
+    switch (event.code) {
+      case KeyCode.Enter:
+      case KeyCode.NumpadEnter:
+      case KeyCode.Tab:
+        this.saveValue();
+        event.preventDefault();
+        return;
+      case KeyCode.Escape:
+        this.cancel.emit();
+        event.preventDefault();
+        event.stopPropagation();
+        return;
+      case KeyCode.ArrowUp:
+      case KeyCode.ArrowDown:
+        event.preventDefault();
+        return;
+    }
+
+    if (DISABLED_CHARACTERS.includes(event.key)) {
```

Comment:
What about case when user paste text with disabled characters here? Is this covered?

Reply:
Good point!